Add helm chart kubernetes deployment

pierreozoux opened the pull request:
@RocketChat/core 

Would love to get your feedback about this PR: https://github.com/kubernetes/charts/pull/752

Also, I think it makes sense to replace openshift folder then.

And finally, solving the following issues would be really nice to have to help this deployment:
 - [ ]  #6236
 - [x] #6235 
 - [x] #6192

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/RocketChat/Rocket.Chat?pullRequest=6340) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/RocketChat/Rocket.Chat?pullRequest=6340) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/RocketChat/Rocket.Chat?pullRequest=6340) it.</sub>

geekgonecrazy commented:
Thanks for starting on this!  Will for sure take a look.  Having started using kubernetes this was in our near future :grin:

Does openshift support helm?  We for sure want to keep openshift support.  We have a lot of users using openshift right now.  So maybe openshift should sit along side helm before we remove it?

pierreozoux commented:
You can use helm on openshift (as it is kubernetes, but maybe not all
openshift users will use helm).

I just find helm so disruptive, that every kubernetes user should use it.
And if not, it is just a better way and up to date of distributing
kubernetes yaml file.

On 13-03-2017 15:36, Aaron Ogle wrote:
> Thanks for starting on this! Will for sure take a look. Having started
> using kubernetes this was in our near future 😁
> 
> Does openshift support helm? We for sure want to keep openshift support.
> We have a lot of users using openshift right now. So maybe openshift
> should sit along side helm before we remove it?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/RocketChat/Rocket.Chat/pull/6340#issuecomment-286144693>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABxvHbALKwxRMF32zl0KKd-GmGzw-51Aks5rlWJjgaJpZM4MbF25>.
> 

-- 
I use PGP to protect our privacy, if you want to know more, you can
follow this
https://emailselfdefense.fsf.org/en/

If you have further questions, please do not hesitate to ask.
You can verify my public key here: https://keybase.io/pierreozoux

geekgonecrazy commented:
I found out about at least a dozen people using it last week alone.  I would prefer to leave it and update our documentation to point to helm, and recommend everyone move before we just yank it.  All for pushing the better way.  But its being used enough I think we have no choice but go through deprecation process.

pierreozoux commented:
I removed the removing of openshift :)

I'll wait the PR to be merged to remove the WIP in the title, and give you the go to merge.

Thanks a lot for your comments in the PR, I addressed them.

geekgonecrazy commented:
Going to go ahead and merge now that in.  I'll write documentation and update the link to point to our documentation

Sing-Li commented:
Awesome!   k8s + RC  💕
